tools/walletextension/api/utils.go

```diff
@@ -46,3 +48,33 @@ func getQueryParameter(params map[string]string, selectedParameter string) (stri
 
 	return value, nil
 }
+
+func getUserID(conn userconn.UserConn, userIDPosition int) (string, error) {
+	// try getting userID from query parameters and return it if successful
+	userID, err := getQueryParameter(conn.ReadRequestParams(), common.UserQueryParameter)
+	if err == nil {
+		if len(userID) != common.MessageUserIDLen {
+			return "", fmt.Errorf(fmt.Sprintf("wrong length of userID from URL. Got: %d, Expected: %d", len(userID), common.MessageUserIDLen))
+		}
+		return userID, err
+	}
+
+	// Alternatively, try to get it from URL path
```

Comment:
pls add a comment here that this is a temporary hack to work around a hard hat bug

Reply:
updated